crates/polars-core/src/chunked_array/ops/arity.rs

```diff
@@ -446,3 +522,123 @@ where
         });
     ChunkedArray::from_chunk_iter(ca1.name(), iter)
 }
+
+pub fn broadcast_binary_elementwise<T, U, V, F>(
+    lhs: &ChunkedArray<T>,
+    rhs: &ChunkedArray<U>,
+    mut op: F,
+) -> ChunkedArray<V>
+where
+    T: PolarsDataType,
+    U: PolarsDataType,
+    V: PolarsDataType,
+    F: for<'a> BinaryFnMut<Option<T::Physical<'a>>, Option<U::Physical<'a>>>,
+    V::Array: for<'a> ArrayFromIter<
+        <F as BinaryFnMut<Option<T::Physical<'a>>, Option<U::Physical<'a>>>>::Ret,
+    >,
+{
```

Comment:
Unfortunately we can only do a fast-path here if both are full-null... Makes me wonder if this is a good idea after all :( Nevertheless, we should do a fast-path here for that case.